### PR TITLE
Page Builder

### PR DESCRIPTION
**Pull request by bryantgillespie**
Main Page Builder PR

### CODE REVIEW

File: site/types.ts [hunk 1]
```diff
@@ -1,28 +1,679 @@
-// Page Builder
-export interface Block {
-	id: string;
-	collection: string;
-	item: object;
+// ** Directus Schema for SDK **
```
Comment (paescuj) [inline site/types.ts:1]:
```suggestion
/** Directus Schema for SDK */
```

File: site/types.ts [hunk 2]
```diff
@@ -1,28 +1,679 @@
...
+	case_studies: CaseStudy[];
+
+	// Meta
+	navigation: Navigation[];
+	globals: Globals[];
+	redirects: Redirect[];
+	seo: SEO[];
+}
+
+// ** Page Builder Types **
```
Comment (paescuj) [inline site/types.ts:26]:
```suggestion
// ** Page Builder Types **

```

File: site/types.ts [hunk 3]
```diff
@@ -1,28 +1,679 @@
...
+	navigation?: string | Navigation | null;
+	has_children?: boolean | null;
+	type?: string | null;
+	open_in_new_tab?: boolean | null;
+	page?: string | Page | null;
+	url?: string | null;
+	parent?: string | Navigation | null;
+	description?: string | null;
+	image?: string | File | null;
+	/** Add this item to far right and make it huge. <strong>Use only 1 per submenu</strong> */
```
Comment (paescuj) [inline site/types.ts:418]:
```suggestion
	/** Add this item to far right and make it huge. **Use only 1 per submenu.** */
```

File: site/types.ts [hunk 4]
```diff
@@ -1,28 +1,679 @@
...
+	button_group?: (string | CompButtonGroup) | null;
+}
+
+export interface BlockHeroRotator {
+	id?: string;
+}
+
+export interface BlockLogocloud {
+	id?: string;
+	/** How do we want to display the logos? */
```
Comment (paescuj) [inline site/types.ts:122]: Is this a TODO?
```suggestion
	/** TODO How do we want to display the logos? */
```

File: site/types.ts [hunk 5]
```diff
@@ -1,28 +1,679 @@
...
+	id?: string;
+	status?: string;
+	sort?: number | null;
+	user_created?: string | User | null;
+	date_created?: string | null;
+	user_updated?: string | User | null;
+	date_updated?: string | null;
+	title?: string | null;
+	seo?: string | SEO | null;
+	/** Start with a slash (/page) and do NOT add a trailing slash (/page/) */
```
Comment (paescuj) [inline site/types.ts:37]:
```suggestion
	/** Start with a slash (/page) and do NOT add a trailing slash (/page/). */
```

File: site/types.ts [hunk 6]
```diff
@@ -1,28 +1,679 @@
...
+	// Meta
+	navigation: Navigation[];
+	globals: Globals[];
+	redirects: Redirect[];
+	seo: SEO[];
+}
+
+// ** Page Builder Types **
+export interface Page {
+	id?: string;
```
Comment (paescuj) [inline site/types.ts:28]: Shouldn't the `id` field (and probably other fields) be given for almost all types?

**Comment by bryantgillespie** [comment]
> @bryantgillespie Does this superseed #18?

This was built from that one so I guess it does supersede it. Feel free to close #18

Didn't realize it was only types I put into this one.

File: site/types.ts (hunk 6 above)
Comment (rijkvanzanten) [reply site/types.ts]: Yeah all properties in here shouldn't be optional. The SDK will `Pick<>` what's requested, and everything else is `x | null` typed 👍🏻

**Review by rijkvanzanten — APPROVED** [review]

File: site/types.ts (hunk 6 above)
Comment (paescuj) [reply site/types.ts]: Perfect 👍🏻 Thanks for updating 🙏

**Pull request merged**